src/test/java/com/iabtcf/encoder/field/IntEncoderTest.java

```diff
@@ -0,0 +1,37 @@
+package com.iabtcf.encoder.field;
+
+import org.junit.Assert;
+import org.junit.Test;
+
+public class IntEncoderTest {
+    private static final String VALUE_1 = Integer.toBinaryString(1);
+    private static final String VALUE_10 = Integer.toBinaryString(10);
+    private static final Integer EXPECTED_VALUE_1 = 1;
+    private static final Integer EXPECTED_VALUE_10 = 10;
+
+    private static final String INVALID_VALUE = "invalid_value";
+    private static final String EMPTY_VALUE = "";
+    private static final String NULL_VALUE = null;
+
+    @Test
+    public void whenValueIsValidThenShouldReturnExpectedValue() {
+        Assert.assertEquals(EXPECTED_VALUE_1, IntEncoder.getInstance().decode(VALUE_1));
+        Assert.assertEquals(EXPECTED_VALUE_10, IntEncoder.getInstance().decode(VALUE_10));
+    }
+
+    @Test
+    public void whenValueIsInvalidShouldReturnNull() {
+        Assert.assertNull(IntEncoder.getInstance().decode(INVALID_VALUE));
```

Comment:
idem (if language feature – don't test)

Reply:
But if in the future the implementation of IntDecoder changes? We need to ensure that it continues to return null regardless of how it is implemented.

Reply:
I'm not sure I follow, I don't think it should ever return `null`.

Reply:
I will change this.